plugins/inputs/googlecoreiot/googlecoreIot_test.go

```diff
@@ -0,0 +1,333 @@
+package googlecoreiot
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	influx "github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testMsg          = `{"message":{"attributes":{"deviceId":"myPi","deviceNumId":"2808946627307959","deviceRegistryId":"my-registry","deviceRegistryLocation":"us-central1","projectId":"conference-demos","subFolder":""},"data":"dGVzdGluZ0dvb2dsZSxzZW5zb3I9Ym1lXzI4MCB0ZW1wX2M9MjMuOTUsaHVtaWRpdHk9NjIuODMgMTUzNjk1Mjk3NDU1MzUxMDIzMQ==","messageId":"204004313210337","message_id":"204004313210337","publishTime":"2018-09-14T19:22:54.587Z","publish_time":"2018-09-14T19:22:54.587Z"},"subscription":"projects/conference-demos/subscriptions/my-subscription"}\n`
```

Comment:
This test file need pretty significant changes.  The test data needs to be created to exercise specific functionality, and can't just be a big pile of actual data, as it makes it impossible to tell what is being tested in each case.  Each test case should test one piece of functionality in googleiotcore.go, if you have lots of similar cases use a table test.  On the testmsgs.gz, we should just gzip the body before sending it to the test function, in code instead of precompressed on disk.  We can talk more about this directly.